examples/profile.js

```diff
@@ -0,0 +1,52 @@
+goog.provide('profile_example');
+
+
+goog.require('ngeo');
+goog.require('ngeo.profileDirective');
+
+
+/** @const **/
+var app = {};
+
+
+/** @type {!angular.Module} **/
+app.module = angular.module('app', ['ngeo']);
+
+
+
+/**
+ * @constructor
+ * @param {angular.$http} $http The $http angular service.
+ * @param {angular.Scope} $scope The $scope angular service.
+ */
+app.MainController = function($http, $scope) {
+
+  $http.get('data/profile.json').then(angular.bind(this, function(resp) {
+    this['profileData'] = resp.data;
+  }));
+
+  this['point'] = null;
+
+  this.scope_ = $scope;
+};
+
+
+/**
+ * @param {Object} point The point object with x, y, elevation, distance
+ * information.
+ */
+app.MainController.prototype.onProfileHover = function(point) {
+  this['point'] = point;
+  this.scope_.$apply();
```

Comment:
Yes, for the controller the `onProfileHover` function is executed in the
context of Angular, so the controller should not have to call $apply. You
can try this in the directive:

```js
profile.onHover(function() {
  scope.$apply(function() {
    scope.$eval(attrs['ngeoProfileOnhover'], {'point': point});
  });
});
```




On Fri, Feb 27, 2015 at 2:10 PM, Pierre GIRAUD <notifications@github.com>
wrote:

> In examples/profile.js
> <https://github.com/camptocamp/ngeo/pull/173#discussion_r25504857>:
>
> > +    this['profileData'] = resp.data;
> > +  }));
> > +
> > +  this['point'] = null;
> > +
> > +  this.scope_ = $scope;
> > +};
> > +
> > +
> > +/**
> > + * @param {Object} point The point object with x, y, elevation, distance
> > + * information.
> > + */
> > +app.MainController.prototype.onProfileHover = function(point) {
> > +  this['point'] = point;
> > +  this.scope_.$apply();
>
> This seems to be required. Should I do it in the directive?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/camptocamp/ngeo/pull/173/files#r25504857>.
>



-- 
Eric Lemoine

Camptocamp France SAS
Savoie Technolac, BP 352
73377 Le Bourget du Lac, Cedex

Tel : 00 33 4 79 44 44 94
Mail : eric.lemoine@camptocamp.com
http://www.camptocamp.com